### PR TITLE
Implement URL validation on default videos links

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?
Adds validation to default links to prevent miss use 

https://www.loom.com/share/a27229dba13747a29bcd249e0c320453

### CODE REVIEW

File: packages/app-store/locations.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export type DefaultEventLocationType = {
   category: string;
 
   iconUrl: string;
-
+  urlRegExp?: string;
```
Comment (sean-brydon) [inline packages/app-store/locations.ts:18]: This existed in the object just wasnt typed

File: packages/trpc/server/routers/viewer.tsx [hunk 2]
```diff
@@ -1192,6 +1192,24 @@ const loggedInViewerRouter = router({
     )
     .mutation(async ({ ctx, input }) => {
       const currentMetadata = userMetadata.parse(ctx.user.metadata);
+      const credentials = ctx.user.credentials;
+      const foundApp = getApps(credentials).filter((app) => app.slug === input.appSlug)[0];
+
+      if (!foundApp) throw new TRPCError({ code: "BAD_REQUEST", message: "App not found" });
```
Comment (sean-brydon) [inline packages/trpc/server/routers/viewer.tsx:1198]: Added a bit more validation so you cant break things

Comment (hariombalhara) [reply packages/trpc/server/routers/viewer.tsx:1198]: Given the condition, I think the error message should probable be "App not installed"

Comment (sean-brydon) [reply packages/trpc/server/routers/viewer.tsx:1198]:
```suggestion
      if (!foundApp) throw new TRPCError({ code: "BAD_REQUEST", message: "App not installed" });
```

File: packages/trpc/server/routers/viewer.tsx [hunk 3]
```diff
@@ -1192,6 +1192,24 @@ const loggedInViewerRouter = router({
...
+
+      if (foundApp.appData?.location?.linkType === "static" && !input.appLink) {
+        throw new TRPCError({ code: "BAD_REQUEST", message: "App link is required" });
+      }
+
+      if (foundApp.appData?.location?.linkType === "static" && foundApp.appData.location.urlRegExp) {
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer.tsx:1204]: nit: Maybe store  `foundApp.appData?.location?` in a variable

**Review by hariombalhara — APPROVED** [review]

**Review by JeroenReumkens — APPROVED** [review]
Great work!

**Pull request merged**